test/cmocka/src/math/trig/sin_fixed.c

```diff
@@ -121,7 +121,7 @@ static void test_math_trig_sin_fixed(void **state)
 		double rad = M_PI / (180.0 / theta);
 		int32_t rad_q28 = Q_CONVERT_FLOAT(rad, 28);
 
-		float r = Q_CONVERT_QTOF(sin_fixed(rad_q28), 31);
+		float r = Q_CONVERT_QTOF(sin_fixed(rad_q28), 30);
```

Comment:
does this function replacement change the Q format returned?

Reply:
Yes, From Q1.31 to Q2.30. 
I missed to change it here since _int32_t sin_fixed(int32_t th_rad_fxp)_ is Q2.30 . As a result cmocka test failed . 
Please see figure -3 , plot displays difference between floating and fixed point sine for all 360 degree values

UT test will fail if floating to fixed result difference is greater than defined tolerance value of 0. 000005

Reply:
hmm did we check to see if any users need to be adjusted who use this function. If yes we need to add additional commits to make them compatible

Reply:
Yes - this is needed
drc
_int32_t abs_sin_val = sin_fixed(q_mult(ABS(x), PI_OVER_TWO, 30, 30, 28));_
Tone generator
_sine = q_mults_32x32(sin_fixed(sg->w), sg->a, Q_SHIFT_BITS_64(31, 31, 31));_

Reply:
> Look pretty good, can you check the things mentioned by me and Keyon!

Added the changes as per required change.

> Look pretty good, can you check the things mentioned by me and Keyon!

Thanks a lot for review and inputs.